File: backend/src/main/java/ua/textlink/auth/entity/TextBlock.java
package ua.textlink.auth.entity;

import jakarta.persistence.*;

import java.time.Instant;

@Entity
@Table(
        name = "text_blocks",
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_text_blocks_slug",
                        columnNames = "slug"
                )
        }
)
public class TextBlock {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 12)
    private String slug;

    @Column(length = 255)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String text;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "expires_at", nullable = false)
    private Instant expiresAt;

    @Column(name = "is_active", nullable = false)
    private boolean active = true;

    @Column(name = "views_count", nullable = false)
    private long viewsCount = 0;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
            name = "owner_id",
            nullable = false,
            foreignKey = @ForeignKey(
                    name = "fk_text_blocks_owner"
            )
    )
    private UserAccount owner;

    @PrePersist
    void prePersist() {
        if (createdAt == null) {
            createdAt = Instant.now();
        }
    }

    public Long getId() {
        return id;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(String slug) {
        this.slug = slug;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(Instant expiresAt) {
        this.expiresAt = expiresAt;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public long getViewsCount() {
        return viewsCount;
    }

    public void setViewsCount(long viewsCount) {
        this.viewsCount = viewsCount;
    }

    public UserAccount getOwner() {
        return owner;
    }

    public void setOwner(UserAccount owner) {
        this.owner = owner;
    }
}

File: frontend/src/types/textBlock.ts
export type CreateBlockRequest = {
  title: string;
  text: string;
  expirationMinutes: number;
};

export type UpdateBlockRequest = {
  title: string;
  text: string;
  expirationMinutes?: number;
};

export type TextBlockResponse = {
  slug: string;
  publicUrl: string;
  expiresAt: string;
};

export type TextBlockView = {
  slug: string;
  title: string;
  text: string;
  createdAt: string;
  expiresAt: string;
  viewsCount: number;
  active: boolean;
  canManage: boolean;
};

export type TextBlockEdit = {
  slug: string;
  title: string;
  text: string;
  createdAt: string;
  expiresAt: string;
  active: boolean;
};

export type MyTextBlock = {
  slug: string;
  title: string;
  publicUrl: string;
  createdAt: string;
  expiresAt: string;
  viewsCount: number;
  active: boolean;
};